### PR TITLE
add manifest file reader/writer.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @ansyral @vicancy @hellosnow @qinezh @superyyrrzz

### CODE REVIEW

File: src/Microsoft.DocAsCode.Common/FileAbstractLayer/ManifestFileReader.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            return new PathMapping(file, entry.LinkToPath ?? Path.Combine(ManifestFolder, entry.RelativePath));
+        }
+
+        public IEnumerable<RelativePath> EnumerateFiles()
+        {
+            lock (_syncRoot)
+            {
+                return (from f in Manifest.Files
+                        from ofi in f.OutputFiles.Values
+                        select ((RelativePath)ofi.RelativePath).GetPathFromWorkingFolder()).ToList();
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Common/FileAbstractLayer/ManifestFileReader.cs]: >ToList() [](start = 92, length = 8)

do we need Distinct() here, for example, redirection?

**Comment by ansyral** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**